cfme/tests/cloud_infra_common/test_retirement.py

```diff
@@ -9,37 +9,34 @@
 from utils.wait import wait_for
 
 
+pytestmark = [
+    pytest.mark.usefixtures('uses_infra_providers', 'uses_cloud_providers')
+]
+
+
 def pytest_generate_tests(metafunc):
     # Filter out providers without provisioning data or hosts defined
-    argnames, argvalues, idlist = testgen.infra_providers(
-        metafunc, 'provisioning', template_location=["provisioning", "template"])
+    argnames, argvalues, idlist = testgen.all_providers(
+        metafunc, 'small_template', template_location=["small_template"])
 
-    new_idlist = []
-    new_argvalues = []
-    for i, argvalue_tuple in enumerate(argvalues):
-        args = dict(zip(argnames, argvalue_tuple))
-        if not args['provisioning']:
-            # No provisioning data available
-            continue
+    testgen.parametrize(metafunc, argnames, argvalues, ids=idlist, scope="module")
 
-        # required keys should be a subset of the dict keys set
-        if not {'template', 'host', 'datastore'}.issubset(args['provisioning'].viewkeys()):
-            # Need all three for template provisioning
-            continue
 
-        new_idlist.append(idlist[i])
-        new_argvalues.append(argvalues[i])
-
-    testgen.parametrize(metafunc, argnames, new_argvalues, ids=new_idlist, scope="module")
+@pytest.fixture(scope="function")
+def vm_crud(provider, small_template):
+    return VM.factory(
+        'test_retire_prov_{}'.format(fauxfactory.gen_alpha(length=8).lower()),
+        provider, template_name=small_template)
```

Comment:
Is there a need to split these two up here? To me it is less confusing to keep the crud retrieval inside the ```vm``` fixture.